### PR TITLE
Fixing PR #1804

### PR DESCRIPTION
**Pull request by svanharmelen**
@mitchellh looking at PR #1811 I think this one also needs to be updated. Not a 100% sure, but I would think we don't need to add the `SharedCredentialsProvider` and `EC2RoleProvider` for this one right?

### CODE REVIEW

**Comment by mitchellh** [comment]
AH, we might as well add those, it doesn't hurt. Do you mind?

**Comment by svanharmelen** [comment]
Sure, no problem... Coming up...

**Comment by svanharmelen** [comment]
Added `SharedCredentialsProvider` and `EC2RoleProvider` as well.

File: state/remote/s3.go [hunk 1]
```diff
@@ -26,19 +26,30 @@ func s3Factory(conf map[string]string) (Client, error) {
...
 
 	accessKeyId := conf["access_key"]
 	secretAccessKey := conf["secret_key"]
 
-	credentialsProvider := credentials.NewStaticCredentials(accessKeyId, secretAccessKey, "")
+	credentialsProvider := credentials.NewChainCredentials([]credentials.Provider{
+		&credentials.StaticProvider{Value: credentials.Value{
+			AccessKeyID:     accessKeyId,
+			SecretAccessKey: secretAccessKey,
+			SessionToken:    "",
```
Comment (radeksimko) [inline state/remote/s3.go]: There is actually a token config option, I see no reason to set this to empty string.

Comment (radeksimko) [inline state/remote/s3.go]: Ah, forgive me + ignore me, this is a config for aws provider...

**Comment by mitchellh** [comment]
LGTM

**Pull request merged**